trinity/extensibility/plugin.py

```diff
@@ -85,35 +90,41 @@ def name(self) -> str:
     def logger(self) -> logging.Logger:
         return logging.getLogger('trinity.extensibility.plugin.BasePlugin#{0}'.format(self.name))
 
+    @property
+    def event_bus(self) -> Endpoint:
+        if self.context is None:
+            raise EventBusNotReady("Tried accessing ``event_bus`` before ``ready`` was called")
+
+        return self.context.event_bus
+
     def set_context(self, context: PluginContext) -> None:
         """
         Set the :class:`~trinity.extensibility.plugin.PluginContext` for this plugin.
         """
         self.context = context
 
-    def configure_parser(self, arg_parser: ArgumentParser, subparser: _SubParsersAction) -> None:
+    def ready(self) -> None:
         """
-        Called at startup, giving the plugin a chance to amend the Trinity CLI argument parser
+        Called after the plugin received its context and is ready to bootstrap itself.
```

Comment:
It'd be cool to have a large docstring for this class which diagram'd the lifecycle of a plugin (and probably a page in the docs once the API stablizes.)

Reply:
Yes, I plan to improve docs for plugins soon! (https://github.com/ethereum/py-evm/issues/1103)